packages/react-core/src/components/Select/SelectMenu.tsx

```diff
@@ -145,35 +149,27 @@ export class SelectMenu extends React.Component<SelectMenuProps> {
       ...props
     } = this.props;
     /* eslint-enable @typescript-eslint/no-unused-vars */
+    const overflowProps =
+      maxWidth || maxHeight
+        ? { style: { width: maxWidth, maxHeight, overflow: 'auto', minWidth: maxWidth ? 'auto' : undefined } }
```

Comment:
If an invalid `maxWidth` is used, the menu still gets `overflow: auto` and `min-width: auto`. Is that intentional? It still changes the presentation of the menu in that case.

We can always add this support in core and provide a variable for the react component (or a user) to customize.

Reply:
It would be a bit hard to check for valid values, I think this kind of check isn't even done by CSS-in-JS libs, is it?

And yeah, support for this on the core css-with a modifier I guess?-would be cool.

Reply:
> And yeah, support for this on the core css-with a modifier I guess?-would be cool.

That's how I would like to approach it if we can!  Same for the `maxHeight` prop.

We could have something like

```
.pf-m-scrollable/contrain-height {
    max-height: var(--pf-c-select__menu--MaxHeight); // defaults to something reasonable specified by design. Maybe 300px?
    overflow: auto;
}

.pf-m-contrain-width {
    width: auto;
    max-width: var(--pf-c-select__menu--MaxWidth); // defaults to auto maybe? The menu would be the size of the widest item in the menu
    overflow: auto;
}
```

The component could support a new prop (`scrollable/constrainHeight`) that would only apply the `pf-m-scrollable/constrain-height` modifier. And it would automatically apply `.pf-m-scrollable/constrain-height` if `maxHeight` is used, and use something like `.setProperty('--pf-c-select__menu--MaxHeight', maxHeight)`, to not break the API.

Same for `.pf-m-constrain-width`. wdyt? cc @tlabaj

Reply:
What about backward compatibility for the `maxHeight` prop? I guess we could keep the prop to override the `--pf-c-select__menu--MaxHeight` var you proposed, which I guess it's the same thing `scrollable/constrainHeight` would do?

Reply:
Makes sense. How do you want to proceed? Should I open a pr to add the modifiers on the PatternFly repo?

Reply:
When are you aiming to have this feature implemented?

Reply:
Sorry got distracted by other stuff. I'm checking the `patternfly` repo dev process, I have to digest it a bit. Feel free to jump in for this if you want :D

Reply:
I'm happy to do it! I was curious when you're trying to get this feature by? We have a release on June 24, but it may be hard to get this feature in by then. It may be more realistic to target this in for the release after that, which is in the middle of July. Would that be OK?

Reply:
Ah, sure that would work, I have no urgent need. On Atlasmap I applied a workaround but I figured it could be a common enough case to open a PR.